nixpkgs_review/nix.py

```diff
@@ -246,17 +247,30 @@ def nix_build(
     build = cache_directory.joinpath("build.nix")
     write_shell_expression(build, filtered, system)
 
-    command = [
-        "nix",
-        "--extra-experimental-features",
-        "nix-command" if allow.url_literals else "nix-command no-url-literals",
-        "build",
-        "--no-link",
-        "--keep-going",
-        "--allow-import-from-derivation"
-        if allow.ifd
-        else "--no-allow-import-from-derivation",
-    ]
+    if nom:
```

Comment:
this whole command does not need to be duplicated, `nom` should have a compatible CLI and the only thing we need to change is `"nix"`

Reply:
Thats what I did first. It does not work, because `nom` expects all build arguments after `build`.  This is why the order of arguments is slightly changed.

Reply:
`nix` should be able to handle build arguments after `build`, we can change the order of the `nix` invocations to be the same as `nom`'s